src/components/AttachmentModal.js

```diff
@@ -42,28 +45,98 @@ const propTypes = {
 };
 
 const defaultProps = {
-    title: '',
-    sourceURL: null,
-    onConfirm: null,
-    isAuthTokenRequired: false,
+    sortedReportActions: [],
+    visibleAttachment: {
+        sourceURL: null,
+        isAttachment: false,
+        isModalOpen: false,
+        file: {name: ''},
+    },
 };
 
 class AttachmentModal extends Component {
     constructor(props) {
         super(props);
 
+        this.handleKeyPress = this.handleKeyPress.bind(this);
+        this.getNextAttachment = this.getNextAttachment.bind(this);
+        this.getAttachmentSource = this.getAttachmentSource.bind(this);
+
         this.state = {
-            isModalOpen: false,
-            file: null,
-            sourceURL: props.sourceURL,
+            title: 'Attachment',
         };
     }
 
+    componentDidUpdate(prevProps) {
+        const {visibleAttachment: {isModalOpen}} = this.props;
+        if (!prevProps.visibleAttachment.isModalOpen && isModalOpen) {
+            document.addEventListener('keydown', this.handleKeyPress, false);
+        } else if (!isModalOpen) {
+            document.removeEventListener('keydown', this.handleKeyPress, false);
+        }
+    }
+
+    // Using regex to get the src attribute from the html img tag
+    getAttachmentSource(htmlString) {
+        const imgRegx = /<img.*?data-expensify-source="(.*?)"[^>]+>/g;
+        const imageGroup = imgRegx.exec(htmlString);
+        return imageGroup[1];
+    }
+
+    /**
+     * A function to get the next attachment based on the current attachment being show in the report.
+     *
+     * @param {Boolean} toRight
+     */
+    getNextAttachment = (toRight) => {
+        const {sortedReportActions, visibleAttachment: {sourceURL}} = this.props;
+        const attachments = _.filter(sortedReportActions, sortedReportAction => sortedReportAction.action.isAttachment);
+        if (attachments.length <= 1) {
+            return;
+        }
+
+        // Finding the current attachment inside the report using the sourceURL
+        const currentAttachment = attachments.find((attachment) => {
+            const html = lodashGet(attachment, ['action', 'message', 0, 'html'], '');
+            return html.includes(sourceURL);
+        });
+        if (!currentAttachment) {
+            return;
+        }
+        const actions = [...sortedReportActions].reverse();
+        const sequenceNumber = currentAttachment.action.sequenceNumber;
+        const actionsToSearch = toRight
+            ? actions.slice(sequenceNumber)
+            : actions.slice(0, sequenceNumber !== 0 ? sequenceNumber - 1 : 0).reverse();
+        let nextAttachment = actionsToSearch.find(actionResult => actionResult.action.isAttachment);
```

Comment:
This logic here seems pretty complicated. I think you can achieve the same results on this method using just a few lines of the underscore [chain command](https://underscorejs.org/#chain). So basically the whole logic in this method could be simplified to something like this,
```jsx
const currentSequenceNumber = this.props.visibleAttachment.sequenceNumber;
const nextAttachmentAction = _.chain(this.props.sortedReportActions)
    .pluck('action')
    .filter(action => (toRight ? action.sequenceNumber > currentSequenceNumber : action.sequenceNumber < currentSequenceNumber))
    .find(actions => actions.isAttachment)
    .value();
```

Reply:
As I mentioned above:

`I read the requirement saying "The attachments present in the chat should cycle" literally. If there is a consensus that its overkill, I can remove it.`

Reply:
My apologies, I talked this over with @chiragsalian and I think he makes a good point that the carousel (i.e. cycling) behavior is unnecessary in this context, as there is unlikely to be a context where you need to loop back to the first image you ever uploaded. i think we can probably cut this behavior.

Reply:
Well the example i posted is just a way to clean up the logic to make it easier to read. Like i mentioned, in untested which also means you can tweak it as needed. The idea is using chain makes it read. For example if carousel was needed the logic would be something like this, (untested)
```jsx
const nextReportActions = // slice reports actions in the correct direction to the end or beginning based on direction.
const carouselReportActions = // these are the reports actions that loop and we can slice on the opposing direction and reverse the list.

// Then we'd use chain again to do
const nextAttachmentAction = _.chain(nextReportActions)
    .union(carouselReportActions)
    .pluck('action')
    .filter(action => (toRight ? action.sequenceNumber > currentSequenceNumber : action.sequenceNumber < currentSequenceNumber))
    .find(actions => actions.isAttachment)
    .value();
```

But anyway from this [comment](https://github.com/Expensify/Expensify.cash/issues/1038#issuecomment-759716134) it looks like the carousel is not needed so can you still clean up the logic using chain.

Reply:
ok